MoreLinq.NoConflictGenerator/Program.cs

```diff
@@ -0,0 +1,227 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#endregion
+
+namespace MoreLinq.NoConflictGenerator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text.RegularExpressions;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+    static class Program
+    {
+        static void Run(IEnumerable<string> args)
+        {
+            var dir = Directory.GetCurrentDirectory();
+
+            string includePattern = null;
+            string excludePattern = null;
+            var usings = new List<string>();
+
+            Exception MissingArgValue() =>
+                new InvalidOperationException("Missing argument value.");
+
+            using (var arg = args.GetEnumerator())
+            {
+                while (arg.MoveNext())
+                {
+                    switch (arg.Current)
+                    {
+                        case "-i":
+                        case "--include":
+                            includePattern = Read(arg, MissingArgValue);
+                            break;
+                        case "-x":
+                        case "--exclude":
+                            excludePattern = Read(arg, MissingArgValue);
+                            break;
+                        case "-u":
+                        case "--using":
+                            usings.Add(Read(arg, MissingArgValue));
+                            break;
+                        case "":
+                            continue;
+                        default:
+                            dir = arg.Current[0] != '-'
+                                ? arg.Current
+                                : throw new Exception("Invalid argument: " + arg.Current);
+                            break;
+                    }
+                }
+            }
+
+            var includePredicate
+                = string.IsNullOrEmpty(includePattern)
+                ? (_ => true)
+                : new Func<string, bool>(new Regex(includePattern).IsMatch);
+
+            var excludePredicate
+                = string.IsNullOrEmpty(excludePattern)
+                ? (_ => false)
+                : new Func<string, bool>(new Regex(excludePattern).IsMatch);
+
+            var thisAssemblyName = typeof(Program).GetTypeInfo().Assembly.GetName();
+
+            var q =
+                from fp in Directory.EnumerateFiles(dir, "*.cs")
+                select new
+                {
+                    SourcePath = fp,
+                    Methods =
+                        from cd in
```

Comment:
This feels a bit too large for a nested query. I'd factor it out to a separate method. This is more of an aesthetic preference though so feel free to ignore.